karma.conf.js

```diff
@@ -0,0 +1,100 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/0.13/config/configuration-file.html
+
+module.exports = function(config) {
+  config.set({
+    basePath: '../',
+    frameworks: ['jasmine', '@angular-devkit/build-angular'],
+    plugins: [
+      require('karma-jasmine'),
+      require('karma-browserstack-launcher'),
+      require('karma-chrome-launcher'),
+      require('karma-coverage-istanbul-reporter'),
+      require('karma-jasmine-html-reporter'),
+      require('@angular-devkit/build-angular/plugins/karma'),
+      require('karma-scss-preprocessor'),
+      require('karma-spec-reporter')
+    ],
+    client: {
+      clearContext: false // leave Jasmine Spec Runner output visible in browser
+    },
+    coverageIstanbulReporter: {
+      dir: require('path').join(__dirname, 'coverage'),
+      reports: ['html', 'lcovonly'],
+      fixWebpackSourcePaths: true
+    },
+    reporters: ['progress', 'kjhtml'],
+    specReporter: {
+      showSpecTiming: true
+    },
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: true,
+    singleRun: false,
+
+    browserStack: {
+      project: 'RADAR-Questionnaire',
+      build: 'Karma Local',
+      username: process.env.BROWSERSTACK_USERNAME,
+      accessKey: process.env.BROWSERSTACK_ACCESS_KEY
+    },
+
+    browsers: ['Chrome'],
+    // browsers: ['Chrome', 'BS_EDGE', 'BS_FIREFOX', 'BS_SAFARI'],
+    customLaunchers: {
+      ChromeDebugging: {
+        base: 'Chrome',
+        flags: ['--remote-debugging-port=9333']
+      },
+      ChromeNoSandbox: {
+        base: 'Chrome',
+        flags: ['--no-sandbox']
+      },
+      BS_CHROME: {
+        base: 'BrowserStack',
+        browser: 'chrome',
+        os: 'Windows',
+        os_version: '10'
+      },
+      BS_FIREFOX: {
+        base: 'BrowserStack',
+        browser: 'firefox',
+        os: 'Windows',
+        os_version: '10'
+      },
+      BS_SAFARI: {
+        base: 'BrowserStack',
+        browser: 'safari',
+        os: 'OS X',
+        os_version: 'El Capitan'
+      },
+      BS_EDGE: {
+        base: 'BrowserStack',
+        browser: 'edge',
+        os: 'Windows',
+        os_version: '10'
+      },
+      BS_IOS9: {
+        base: 'BrowserStack',
+        device: 'iPad Mini 4',
+        os: 'ios',
+        os_version: '9.1'
+      }
+    },
+    browserDisconnectTimeout: 20000,
+    browserDisconnectTolerance: 1,
+    browserNoActivityTimeout: 240000,
+    captureTimeout: 240000
+  })
+
+  if (process.env['TRAVIS']) {
```

Comment:
what is this used for?

Reply:
This is for integration with `Browserstack`. I was making it similar to the dashboard which had Browserstack integrated. I guess I can remove it in the meantime and put it back when it's fully integrated?

Reply:
I think it's ok to keep it if it is required.

Reply:
I just asked because bettercodehub fails as karma.conf.js file is too big

Reply:
I just removed it since I'll work on it next time and will be adding the specific tests first. 😅

Reply:
Ok cool. No problem